hack/testing-olm-upgrade/test-020-olm-upgrade.sh

```diff
@@ -0,0 +1,108 @@
+
+log::info "Deploying ES secrets..."
+for file in "${repo_dir}"/hack/testing-olm/secrets/*.yaml; do
+  oc apply -f "${file}" -n openshift-operators-redhat
+done;
+#log::info "Creating test namespace: ${NAMESPACE}"
+#oc create namespace "${NAMESPACE}"
+
+# deploy elasticsearch CR
+log::info "Deployng ES CR..."
+oc -n "openshift-operators-redhat" create -f ${repo_dir}/hack/cr.yaml
+
+# check if there are elasticsearch pod
+log::info "Checking if ES pod is ready in the namespace openshift-operators-redhat..."
+try_until_text "oc -n openshift-operators-redhat get pods -l component=elasticsearch -o jsonpath={.items[0].status.phase} --ignore-not-found" "Running" ${TIMEOUT_MIN}
+
+log::info "Getting the previous statate of elasticsearch-operator deployment"
+# get the previous status of the elasticsearch-operator
+oc describe -n "openshift-operators-redhat" deployment/elasticsearch-operator > "$ARTIFACT_DIR"/elasticsearch-operator.describe.before_update 2>&1
+
+log::info "Deploying the ES operator from the catalog..."
+# deploy cluster logging catalog from local code
+"${repo_dir}"/olm_deploy/scripts/catalog-deploy.sh
+
+log::info "Patching subscription..."
+# patch subscription
+payload="{\"op\":\"replace\",\"path\":\"/spec/source\",\"value\":\"elasticsearch-catalog\"}"
+payload="$payload,{\"op\":\"replace\",\"path\":\"/spec/sourceNamespace\",\"value\":\"openshift-operators-redhat\"}"
+payload="$payload,{\"op\":\"replace\",\"path\":\"/spec/channel\",\"value\":\"$version\"}"
+oc -n "openshift-operators-redhat" patch subscription elasticsearch-operator --type json -p "[$payload]"
+
+# patch the minikube version
+log::info "Patching minKubeVersion to 1.16.1..."
+pl="{\"op\":\"replace\",\"path\":\"/spec/minKubeVersion\",\"value\":\"1.16.1\"}"
+try_until_success "oc -n openshift-operators-redhat patch clusterserviceversion elasticsearch-operator.v${version}.0 --type json -p [$pl]" ${TIMEOUT_MIN}
+
+#verify deployment is rolled out
+log::info "Checking if deployment successfully updated..."
+OPENSHIFT_VERSION=${OPENSHIFT_VERSION:-4.5}
+IMAGE_ELASTICSEARCH_OPERATOR=${IMAGE_ELASTICSEARCH_OPERATOR:-registry.svc.ci.openshift.org/ocp/${OPENSHIFT_VERSION}:elasticsearch-operator}
+if [ -n "${IMAGE_FORMAT:-}" ] ; then
+  IMAGE_ELASTICSEARCH_OPERATOR=$(echo $IMAGE_FORMAT | sed -e "s,\${component},elasticsearch-operator,")
+fi
+log::info "Checking if deployment version is ${IMAGE_ELASTICSEARCH_OPERATOR}..."
+try_until_text "oc -n openshift-operators-redhat get deployment elasticsearch-operator -o jsonpath={.spec.template.spec.containers[0].image}" "${IMAGE_ELASTICSEARCH_OPERATOR}" ${TIMEOUT_MIN}
+
+log::info "Checking if the ES pod is running..."
+try_until_text "oc -n openshift-operators-redhat get deployment -l component=elasticsearch -o jsonpath={.items[0].status.updatedReplicas} --ignore-not-found" "1" ${TIMEOUT_MIN}
```

Comment:
is there more we can do/check to confirm that ES is fully upgraded and healthy?  this seems like a very minimal smoketest.

Reply:
once ES is running, we should be able to grab before/after

* indices
* health

Everything should be intact from before.  we likely will have to do some fuzzy logic or add more checks to validate full migration

Reply:
Fixed by adding additional checks for indices and health before and after the upgrade